src/Simulation/CsharpGeneration/RewriteStep.fs

```diff
@@ -54,7 +54,12 @@ type Emitter() =
                 let content = SimulationCode.loadedViaTestNames source context
                 if content <> null then CompilationLoader.GeneratedFile(source, dir, ".dll.g.cs", content) |> ignore
 
+            match Seq.tryExactlyOne compilation.EntryPoints with
+            | Some entryPoint ->
+                let callable = context.allCallables.[entryPoint]
+                let content = EntryPoint.generate context callable
+                CompilationLoader.GeneratedFile(callable.SourceFile, dir, ".EntryPoint.g.cs", content) |> ignore
+            | None -> ()
```

Comment:
Do you want to generate a diagnostic if more than one entry point exists? You should at least return false if several entry point exist.

Reply:
I can do that. Is the thinking that even though the Q# compiler enforces only one entry point, it is still technically possible to give the transformation a syntax tree with more than one?

Reply:
This is now a precondition for the transformation. I did not change the return value of `Transformation` because it assumes that the precondition holds.